### PR TITLE
Fix auto_delete_value of 0 set by wizard

### PR DESCRIPTION
**Pull request by swaterkamp**
**Checklist**:

- [N/A] Tests
- [X] [CHANGELOG](https://github.com/greenbone/gsa/blob/master/CHANGELOG.md) Entry

### CODE REVIEW

**Comment by swaterkamp** [comment]
We are setting that in the model, and I thought the wizard tasks were somehow circumventing that.
Having a second look now, however, it seems like we are checking against '0' instead of 0. That might be the actual problem. Revisiting right now.

**Comment by swaterkamp** [comment]
Indeed that seemed to fix the problem. "Normal" tasks were working before the change to the model, that's what made me think about fixing it in the dialog instead.

**Comment by bjoernricks** [comment]
The models should fix weird or unexpected behavior of our data. The models are intended to get a clean view of the data in our react layer.

File: gsa/src/gmp/models/audit.js [hunk 1]
```diff
@@ -190,7 +190,7 @@ class Audit extends Model {
             break;
           case 'auto_delete_data':
             copy.auto_delete_data =
```
Comment (bjoernricks) [inline gsa/src/gmp/models/audit.js]: Maybe this should be refactored a bit to be independent of the xml parser.
```javascript
const value = parseInt(pref.value);
copy.auto_delete_data = value === 0 ? AUTO_DELETE_KEEP_DEFAULT_VALUE : value;
```

File: gsa/src/gmp/models/task.js [hunk 2]
```diff
@@ -237,7 +237,7 @@ class Task extends Model {
             break;
           case 'auto_delete_data':
             copy.auto_delete_data =
-              pref.value === '0'
+              pref.value === 0
```
Comment (bjoernricks) [inline gsa/src/gmp/models/task.js]: Same as above.

**Pull request merged**